src/components/shared/categoriesModal/CategoriesModal.tsx

```diff
@@ -0,0 +1,83 @@
+import { NamesDBCollection, UserRole } from 'constants';
+
+import { FC } from 'react';
+
+import { Box, Button, Group, Modal, TextInput } from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { useDisclosure } from '@mantine/hooks';
+import { IconPencil } from '@tabler/icons-react';
+import { useTranslation } from 'react-i18next';
+
+import { useAppDispatch, useAppSelector } from 'hooks';
+import {
+  getRole,
+  IProductCategory,
+  setProductCategoryThunk,
+  updateShopDataThunk,
+} from 'store';
+
+type Props = {
+  id: Nullable<string>;
+};
+
+export const CategoriesModal: FC<Props> = ({ id }) => {
+  const { t } = useTranslation();
+
+  const [opened, { open, close }] = useDisclosure(false);
+
+  const dispatch = useAppDispatch();
+
+  const currentUserRole = useAppSelector(getRole);
+
+  const form = useForm({
+    initialValues: {
+      nameCategory: '',
+    },
+  });
+
+  const onSubmitHandler = (values: IProductCategory): void => {
+    if (id) {
+      dispatch(
+        updateShopDataThunk({
+          id,
+          values,
+          NamesDBCollection: NamesDBCollection.ProductCategories,
+        }),
+      );
+    } else {
+      dispatch(setProductCategoryThunk(values));
+    }
+    form.reset();
+  };
+
+  return (
+    currentUserRole === UserRole.Admin && (
```

Comment:
try to create a custom hook for this, so that it checks the role and returns isAdmin, isUser.....

Put it in shared